src/java.base/unix/classes/sun/nio/fs/UnixPath.java

```diff
@@ -90,8 +89,9 @@ static String normalizeAndCheck(String input) {
             checkNotNul(input, c);
             prevChar = c;
         }
-        if (prevChar == '/')
-            return normalize(input, n, n - 1);
+        if (prevChar == '/') {
+            return (n == 1) ? "/" : input.substring(0, n - 1);
+        }
```

Comment:
Looking at it again, I think it would be better if it were changed to
```
        if (prevChar == '/' && n > 1) {
            return input.substring(0, n - 1);
        }
```
That is clearer that it is dropping the trailing slash.